src/components/base/BaseSelect.vue

```diff
@@ -1,43 +1,46 @@
 <script setup lang="ts">
-interface Props {
-  options: string[];
-  modelValue: string;
-}
-const props = withDefaults(defineProps<Props>(), {
-  options: null,
-  modelValue: "",
-});
+const props = withDefaults(
+  defineProps<{
+    options: string[];
+    modelValue: string;
+  }>(),
+  {
+    options: () => [],
+    modelValue: "",
+  }
+);
 </script>
 
 <template>
   <div class="base-select">
-    <label class="base-select__label" for=""><slot /></label>
-    <select
-      class="base-select__select"
-      :modelValue="props.modelValue"
-      @change="
-        $emit('update:modelValue', ($event.target as HTMLInputElement).value)
-      "
-    >
-      <option
-        v-for="option in props.options"
-        :key="option"
-        class="base-select__option"
-        :value="option"
+    <label class="base-select__label" for=""
```

Comment:
So actually, we could handle the `for` attribute really nicely - we could check if the `id` is passed as a prop (and define it as such) and set it here and also as the id of the select.
Either that, or I'd suggest to remove the `for` attribute entirely 😄

Reply:
Point taken. I have passed id and used it to define for attribute.